src/Tribe/Query.php

```diff
@@ -1274,11 +1274,20 @@ public static function getEvents( $args = array(), $full = false ) {
 				}
 
 				$display = Arr::get( $args, 'eventDisplay' );
-				$possible_start_date = Arr::get( $args, 'start_date', Arr::get( $args, 'startDate' ) );
+				$has_date_args = array_filter( [
```

Comment:
why are you using array filter here?

Reply:
To prevent having empty results. Without a callback `array_filter` will filter by empty values.

Reply:
Then it would just work with the conditional you're using next? `! $has_date_args` or you should be checking if empty?

Reply:
No, because if we need to remove the non-existent entries before checking if there is something on `$has_date_args`because otherwise, it would always be true. Since the array would have a few indexes and wouldn't validate that conditional.

Reply:
Yeah, what I mean is that you're using a bool comparison to check an array. After it's filtered. But I assume you've checked already the different cases.